test/E2ETest/Microsoft.Test.E2E.AspNet.OData/Build.AspNetCore3x/BulkOperation/BulkOperationControllerEF.cs

```diff
@@ -0,0 +1,144 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics.Contracts;
+using System.Linq;
+using System.Net;
+using System.Threading;
+using Microsoft.AspNet.OData;
+using Microsoft.AspNet.OData.Extensions;
+using Microsoft.AspNet.OData.Routing;
```

Comment:
BulkOperationControllerEF.cs
BulkOperationPathHandlersEF.cs


is only for Asp.NET Core 3.x?  or it's for ASP.NET Classic and ASP.NET Core?  

if they are for both, you should put "BulkOperation" folder under Micrsoft.Test.E2E.Asp.Net.OData

Reply:
EF uses inmemory db in  this case and its available only for core3 so using it only there, others has other common tests

---
In reply to: [604625776](https://github.com/OData/WebApi/pull/2455#discussion_r604625776) [](ancestors = 604625776)